### PR TITLE
without_review

### PR DESCRIPTION
**Pull request by YaTut1901**

### CODE REVIEW

File: src/main/java/spring/intro/dao/UserDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    public void add(User user) {
+        sessionFactory.openSession().save(user);
+    }
+
+    @Override
+    public List<User> getAll() {
+        return sessionFactory
+                .openSession()
+                .createQuery("from User", User.class)
+                .getResultList();
```
Comment (Martynice) [inline src/main/java/spring/intro/dao/UserDaoImpl.java]: You are better to use try with resources

File: src/main/java/spring/intro/service/UserServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        this.userDao = userDao;
+    }
+
+    @Override
+    public void add(User user) {
+        userDao.add(user);
+    }
+
+    @Override
+    public List<User> usersList() {
```
Comment (RomanDubovskyi) [inline src/main/java/spring/intro/service/UserServiceImpl.java]:
```suggestion
    public List<User> listUsers() {
```

File: src/main/java/spring/intro/dao/UserDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    private SessionFactory sessionFactory;
+
+    @Override
+    public void add(User user) {
+        sessionFactory.openSession().save(user);
+    }
+
+    @Override
+    public List<User> getAll() {
+        return sessionFactory
```
Comment (RomanDubovskyi) [inline src/main/java/spring/intro/dao/UserDaoImpl.java]: do we need to close our resources ?

File: src/main/java/spring/intro/dao/UserDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+@Repository
+public class UserDaoImpl implements UserDao {
+
+    @Autowired
+    private SessionFactory sessionFactory;
+
+    @Override
+    public void add(User user) {
+        sessionFactory.openSession().save(user);
```
Comment (RomanDubovskyi) [inline src/main/java/spring/intro/dao/UserDaoImpl.java]: same here?

File: src/main/java/spring/intro/config/AppConfig.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+import org.springframework.context.annotation.Configuration;
+import org.springframework.orm.hibernate5.LocalSessionFactoryBean;
+import spring.intro.model.User;
+
+@Configuration
+@ComponentScan(basePackages = {
+        "spring.intro.dao",
+        "spring.intro.service"})
+public class AppConfig {
+    public DataSource getDataSource() {
```
Comment (RomanDubovskyi) [inline src/main/java/spring/intro/config/AppConfig.java]: will this work if you try to autowire DataSource anywhere? 
try to test it in main

File: pom.xml [hunk 6]
```diff
@@ -8,5 +8,87 @@
...
+                    <version>3.8.0</version>
+                    <configuration>
+                        <source>${jdk.version}</source>
+                        <target>${jdk.version}</target>
+                        <encoding>${project.build.sourceEncoding}</encoding>
+                    </configuration>
+                </plugin>
+            </plugins>
+        </pluginManagement>
+    </build>
```
Comment (RomanDubovskyi) [inline pom.xml:92]: empty line in the end of the file

**Comment by RomanDubovskyi** [comment]
your travis failed

File: src/main/java/spring/intro/config/AppConfig.java [hunk 7]
```diff
@@ -0,0 +1,39 @@
+package spring.intro.config;
```
Comment (RomanDubovskyi) [inline src/main/java/spring/intro/config/AppConfig.java]:
```suggestion
package spring.project.config;
```

File: src/main/java/spring/intro/dao/UserDaoImpl.java [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+
+import javax.sql.DataSource;
+
+@Repository
+public class UserDaoImpl implements UserDao {
+
+    @Autowired
+    private SessionFactory sessionFactory;
+    @Autowired
+    private DataSource dataSource;
```
Comment (RomanDubovskyi) [inline src/main/java/spring/intro/dao/UserDaoImpl.java]: we don't user dataSource here, let's remove this field

File: src/main/java/spring/intro/dao/UserDaoImpl.java [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+            }
+        }
+    }
+
+    @Override
+    public List<User> getAll() {
+        try (Session session = sessionFactory.openSession()) {
+            return session
+                    .createQuery("from User", User.class)
+                    .getResultList();
```
Comment (RomanDubovskyi) [inline src/main/java/spring/intro/dao/UserDaoImpl.java]:
```suggestion
            return session.createQuery("from User", User.class)
                    .getResultList();
```

File: src/main/java/spring/intro/dao/UserDaoImpl.java [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+import org.springframework.stereotype.Repository;
+import spring.intro.model.User;
+
+import javax.sql.DataSource;
+
+@Repository
+public class UserDaoImpl implements UserDao {
+
+    @Autowired
+    private SessionFactory sessionFactory;
```
Comment (RomanDubovskyi) [inline src/main/java/spring/intro/dao/UserDaoImpl.java]: let's use constructor injection, it's preferrable

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**